pattern_serializer.cpp

```diff
@@ -1,5 +1,8 @@
 #include "pattern_serializer.h"
 #include <cassert>
+#include <string>
+#include <stdio.h>
+#include <stdlib.h>
```

Comment:
why do you need `<stdio.h>` given pattern_serializer already includes the `<cstdio>`?

also, can you use `<cstdlib>` instead of `<stdlib.h>` given that this is C++ and not C?